src/hvm.rs

```diff
@@ -2427,173 +2428,205 @@ pub fn show_rt(rt: &Runtime) -> String {
 }
 
 pub fn show_term(rt: &Runtime, term: Lnk) -> String {
-  let mut lets: HashMap<u128, u128> = HashMap::new();
-  let mut kinds: HashMap<u128, u128> = HashMap::new();
+  enum StackItem {
+    Term(Lnk),
+    Str(String),
+  }
```

Comment:
Why a StackItem can be a Str?

Reply:
This way i can put a string in the stack. This is used to put spaces and closing parenthesis. 
Ex.: In an application: first I put an open parenthesis in the output; then i put a closing parenthesis, the second part of the application, a space and the first part of the application on the stack. In the next iterations the first part of the application will be resolved and will be placed in the output, followed by the space, the second part and the closing parenthesis.